### PR TITLE
Switch to using metamask-compatible codewords

### PR DESCRIPTION
**Pull request by diracdeltas**
fix https://github.com/brave/brave-browser/issues/13245

24: 0xea3C17c81E3baC3472d163b2c8b12ddDAa027874
12: 0x084DCb94038af1715963F149079cE011C4B22961

test plans:

## basic functionality tests

1. create wallet. it should default have 12 seed words. locking and unlocking should produce the same address.
2. import wallet into metamask. it should create the same wallet address.
3. clear extension state. import `cushion pitch impact album daring marine much annual budget social clarify balance rose almost area busy among bring hidden bind later capable pulp laundry` into CW. it should restore address 0xea3C17c81E3baC3472d163b2c8b12ddDAa027874.
5. click "create account" and generate a new wallet. the address should be 0xEc1BB5a4EC94dE9107222c103907CCC720fA3854.
4. import `drip caution abandon festival order clown oven regular absorb evidence crew where` into CW. it should restore address 0x084DCb94038af1715963F149079cE011C4B22961.
5. click "create account" and generate a new wallet. the address should be 0xE60A2209372AF1049C4848B1bF0136258c35f268.

## basic regression tests

do these for both 12 and 24 word wallets.

1. restoring wallet produces the same address
2. adding a new account works
3. importing a new account works
4. locking/unlocking wallet produces the same addresses across all accounts
5. restarting browser produces the same addresses 
6. hardware wallets work
7. transactions and signing work
8. exporting/importing via private key works

## edge case tests

1. create a new 12-word wallet
2. lock the wallet
3. on the unlock screen, click restore.
4. paste in the 24-word phrase.
5. it should restore address `0xea3C17c81E3baC3472d163b2c8b12ddDAa027874`
6. restart browser
7. click restore again, and restore with the 12 word phrase. it should restore address `0x084DCb94038af1715963F149079cE011C4B22961`
8. restart browser
9. repeat step 4 and check that `0xea3C17c81E3baC3472d163b2c8b12ddDAa027874` is restored

### CODE REVIEW

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -1787,7 +1787,7 @@
     "message": "Connect your Ledger hardware wallet to interact with dApps and make transfers to other connected wallets."
   },
   "metamaskImportSubText": {
-    "message": "Note: If importing from a Metamask wallet, enter 12 words instead of 24."
+    "message": "Note: If importing from a Brave wallet created before February 2021, your secret phrase is 24 words instead of 12."
```
Comment (diracdeltas) [inline app/_locales/en/messages.json:1790]: note: `February 2021` should be changed to a release number once we know what release this is landing in

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -188,129 +190,31 @@ export default class MetamaskController extends EventEmitter {
...
-    this.keyringController = new KeyringController({
-      keyringTypes: additionalKeyrings,
-      initState: initState.KeyringController,
-      getNetwork: this.networkController.getNetworkState.bind(this.networkController),
-      encryptor: opts.encryptor || undefined,
-    })
-    this.keyringController.memStore.subscribe((s) => this._onKeyringControllerUpdate(s))
-    this.keyringController.on('unlock', () => this.emit('unlock'))
-
-    this.permissionsController = new PermissionsController({
```
Comment (diracdeltas) [inline app/scripts/metamask-controller.js:201]: note to reviewers: any controllers which depend on `this.keyringController` were moved into a separate function so they could be re-initialized when `this.keyringController` changes to a different KeyringController. the ones that didn't seem to depend on `this.keyringController` were left as-is.

File: app/scripts/background.js [hunk 3]
```diff
@@ -251,17 +251,10 @@ function setupController (initState, initLangCode) {
...
-    debounce(1000),
-    storeTransform(versionifyData),
-    createStreamSink(persistData),
-    (error) => {
-      log.error('MetaMask - Persistence pipeline failed', error)
-    },
-  )
+  onControllerInit(controller)
+  controller.on('controllerInitialized', () => {
+    onControllerInit(controller)
```
Comment (bbondy) [inline app/scripts/background.js:256]: I I don't understand why `onControllerInit` can be called up to twice if `controllerInitialized` is fired. Is that intentional?

**Review by bbondy — COMMENTED** [review]
Tested this out with your test cases and also with a wallet of my own from an old profile, and a new wallet of my own. Everything works well. 1 question in the code though before I can approve.

**Review by bbondy — APPROVED** [review]
> because it needs to initialize the first time a keyringcontroller is created and again when the keyringcontroller is recreated (if needed)
> otherwise the recreated keyringcontroller  will not have its data persisted to disk

**Review by bbondy — APPROVED** [review]
